### PR TITLE
Dact 1690 ch01 correspondence radio controller

### PR DESCRIPTION
**Pull request by jbestwickch**
https://companieshouse.atlassian.net/browse/DACT-1690
Add ch01 correspondence address controller

### CODE REVIEW

**Comment by jbestwickch** [comment]
Not sure why the code coverage is not 100%. The marked code is covered by a parametrised test.

File: src/routes/routes.ts [hunk 1]
```diff
@@ -166,3 +166,6 @@ router.get(urls.UPDATE_DIRECTOR_SUBMITTED, isFeatureEnabled(CH01_ACTIVE), compan
 router.get(urls.UPDATE_DIRECTOR_NATIONALITY, isFeatureEnabled(CH01_ACTIVE), companyAuthenticationMiddleware, checkYourAnswersMiddleware(), updateDirectorNationality.get);
 router.post(urls.UPDATE_DIRECTOR_NATIONALITY, companyAuthenticationMiddleware, nationalityValidator, updateDirectorNationality.post);
 
+router.get(urls.UPDATE_DIRECTOR_CORRESPONDENCE_ADDRESS, isFeatureEnabled(CH01_ACTIVE), companyAuthenticationMiddleware, checkYourAnswersMiddleware(), directorCorrespondenceAddress.get);
```
Comment (AndreKramer7654) [inline src/routes/routes.ts:169]: Are these routes going to the wrong methods? I see by the coverage that the new controller is not getting called.

Comment (AndreKramer7654) [reply src/routes/routes.ts]: That fixed the coverage :)

File: test/controllers/shared.controllers/director.correspondence.address.controller.unit.ts [hunk 2]
```diff
@@ -107,38 +120,43 @@ describe("Director correspondence address controller tests", () => {
...
+        }
+        else{
+          expect(response.text).toContain(UPDATE_DIRECTOR_DETAILS_END);
+        }
       })
     });
 
     describe("post tests", () => {
-      it(`Should render ${DIRECTOR_CORRESPONDENCE_ADDRESS} page if no radio button is selected`, async () => {
+      it.each([PAGE_URL,UPDATE_PAGE_URL])(`Should render ${DIRECTOR_CORRESPONDENCE_ADDRESS} and ${UPDATE_DIRECTOR_CORRESPONDENCE_ADDRESS} page if no radio button is selected`, async () => {
```
Comment (AndreKramer7654) [inline test/controllers/shared.controllers/director.correspondence.address.controller.unit.ts:159]: You need to use the url on line 164

File: test/controllers/shared.controllers/director.correspondence.address.controller.unit.ts [hunk 3]
```diff
@@ -156,7 +174,7 @@ describe("Director correspondence address controller tests", () => {
         expect(mocks.mockCompanyAuthenticationMiddleware).toHaveBeenCalled();
       });
       
-      it("should catch error if patch officer filing failed", async () => {
+      it.each([PAGE_URL,UPDATE_PAGE_URL])("should catch error if patch officer filing failed", async () => {
```
Comment (AndreKramer7654) [inline test/controllers/shared.controllers/director.correspondence.address.controller.unit.ts:177]: use url on line 190

File: src/controllers/shared.controllers/director.correspondence.address.controller.ts [hunk 4]
```diff
@@ -74,17 +77,29 @@ export const postDirectorCorrespondenceAddress = async (req: Request, res: Respo
     const patchFiling = await patchOfficerFiling(session, transactionId, submissionId, officerFilingBody);
 
     if (!canUseRegisteredOfficeAddress && selectedSraAddressChoice === registeredOfficerAddressValue) {
+      // if(isUpdate){
```
Comment (AndreKramer7654) [inline src/controllers/shared.controllers/director.correspondence.address.controller.ts:80]: Better to remove this commented out code and put a note on the future ticket?

**Review by AndreKramer7654 — APPROVED** [review]

**Review by PhilipGlover — APPROVED** [review]
- perhaps it's me, but the descriptions aren't that clear to me; but I gather this is to 'add functionality to update the correspondence address as part of a CH01'
- in which case 👍

**Pull request merged**